crates/red_knot_python_semantic/src/types.rs

```diff
@@ -378,6 +384,12 @@ pub struct IntersectionType<'db> {
     negative: FxOrderSet<Type<'db>>,
 }
 
+#[salsa::interned]
+pub struct StringLiteralType<'db> {
+    #[return_ref]
+    value: String,
```

Comment:
Maybe we could use `Box<str>` here, to save a bit of space, since the data should always be immutable? Though it maybe wouldn't make much difference

Reply:
Great call; this is probably worth doing, considering it saves an entire usize of memory, which is eight ascii characters (on a 64-bit system); it could halve the memory used for a lot of small string literal types.

Reply:
Ah! Yeah, I consistently forget that `Box<str>` is an option. 👍🏼